src/lsp/cobol_parser/expect/gen_expect.ml

```diff
@@ -105,55 +111,57 @@ module SuperSet (OrderedType : Set.OrderedType) = struct
     List.fold_left (fun acc value -> add (f value) acc) init l
 end
 
-module SymbolSet = SuperSet(struct
-  type t = symbol
-  let compare = Symbol.compare
-end)
+module NonterminalSet = Make_set(struct
+    type t = nonterminal
+    let compare = Nonterminal.compare
+  end)
 
 module CompEntrySet = struct
-  include SuperSet(struct
-    type t = completion_entry
-    let compare entry1 entry2 =
-      match entry1, entry2 with
+  include Make_set(struct
+      type t = completion_entry
+      let compare entry1 entry2 =
+        match entry1, entry2 with
         | Custom s1, Custom s2 -> String.compare s2 s1
         | K nel1, K nel2 -> NEL.compare Terminal.compare nel1 nel2
         | Custom _, K _ -> -1
         | K _, Custom _ -> 1
-  end)
+    end)
 end
 
-module Map5 (OrderedType : Map.OrderedType) = struct
+module Make_map (OrderedType : Map.OrderedType) = struct
   include Map.Make(OrderedType)
 
   let add_to_list x data m =
     let add = function None -> Some [data] | Some l -> Some (data :: l) in
     update x add m
 end
 
-module CompEntryMap = Map5(struct
-  type t = CompEntrySet.t
-  let compare = CompEntrySet.compare
-end)
+module CompEntryMap = Make_map(struct
+    type t = CompEntrySet.t
+    let compare = CompEntrySet.compare
+  end)
 
-module ProdMap = Map5(struct
-  type t = Production.t list
-  let compare = List.compare (fun p1 p2 ->
-    Production.to_int p1 - Production.to_int p2)
-end)
+module ProdMap = Make_map(struct
+    type t = Production.t list
+    let compare = List.compare (fun p1 p2 ->
+        Production.to_int p1 - Production.to_int p2)
+  end)
 
-module SymbolMap = Map5(struct
-  type t = SymbolSet.t
-  let compare = SymbolSet.compare
-end)
+module NonterminalMap = Make_map(struct
+    type t = NonterminalSet.t
+    let compare = NonterminalSet.compare
+  end)
```

Comment:
I did not spot: that's more than a typical map. ~~`NonterminalMap`~~/`Nonterminals_mapping` (so the `s` stands a little).